src/main/scala/com/microsoft/hyperspace/index/rules/JoinIndexRule.scala

```diff
@@ -68,18 +72,54 @@ object JoinIndexRule
                     right =
                       RuleUtils.transformPlanToUseIndex(spark, rIndex, r, useBucketSpec = true))
 
+              def getShuffleCnt(sparkPlan: SparkPlan): Long = {
+                sparkPlan.collect { case _: ShuffleExchangeExec => true }.length
+              }
+
+              val resultPlan =
+                if (!HyperspaceConf.hybridScanEnabled(spark) || !HyperspaceConf
+                      .hybridScanShuffleCheckEnabled(spark)) {
+                  updatedPlan
+                } else {
+                  val shuffleCntPair = spark.sessionState
+                    .executePlan(updatedPlan)
+                    .executedPlan
+                    .collect {
+                      case smj: SortMergeJoinExec =>
+                        (getShuffleCnt(smj.left), getShuffleCnt(smj.right))
+                    }
+                  assert(shuffleCntPair.length <= 1)
+                  shuffleCntPair.headOption.flatMap {
+                    // If the number of shuffle is 2, the candidate index pair cannot remove
+                    // the shuffles in both left and right for join and also Hybrid Scan causes
+                    // an additional shuffle for merging appended files. We don't apply the index
```

Comment:
I am a bit confused with this comment. For example, is the additional shuffle by hybrid scan also counting toward 2?

Reply:
1 for sort merge join & 1 for hybrid scan on the fly shuffle, in left or right child plan.

Reply:
Yea please reword the comments.